src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -0,0 +1,218 @@
+                    HighwayTag.TRUNK);
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public RoadNameGapCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && Edge.isMasterEdgeIdentifier(object.getIdentifier())
+                && !HighwayTag.isLinkHighway(object) && VALID_HIGHWAY_TAG.test(object)
+                && !JunctionTag.isRoundabout(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final EdgePredicate edgePredicate = new EdgePredicate(edge);
+        final Set<Edge> inEdges = edge.inEdges().stream().filter(this::validCheckForObject)
+                .filter(edgePredicate::isSameHeading).collect(Collectors.toSet());
+        final Set<Edge> outEdges = edge.outEdges().stream().filter(this::validCheckForObject)
+                .filter(edgePredicate::isSameHeading).collect(Collectors.toSet());
+        if (inEdges.isEmpty() || outEdges.isEmpty())
+        {
+            return Optional.empty();
+        }
+
+        final Set<String> matchingInAndOutEdgeNames = getMatchingInAndOutEdgeNames(inEdges,
+                outEdges);
+
+        if (matchingInAndOutEdgeNames.isEmpty())
+        {
+            // There is no pair of inedge and out edge with same name.
+            return Optional.empty();
+        }
+
+        // Create flag when we have in edge and out edge with same name but intermediate edge
+        // doesn't have a name.
+        if (!edge.getName().isPresent())
+        {
+            return Optional.of(
+                    createFlag(object, this.getLocalizedInstruction(0, edge.getOsmIdentifier())));
+        }
+        final Optional<String> edgeName = edge.getName();
+
+        if (edgeName.isPresent() && !matchingInAndOutEdgeNames.contains(edgeName.get()))
+        {
+            // Case 1: Edge name: Tai. Incoming edge names: Tai , Shai. Outgoing edge name: Shai.
```

Comment:
These are cases we want to flag, right? If so, we can probably indicate that and I'd vote to move to the docs

Reply:
Moved it to RoadNameGapCheck.md file case examples.